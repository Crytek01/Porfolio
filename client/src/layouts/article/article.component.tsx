import { FunctionComponent } from 'react';

import { ArticleContainer } from './article.styles';

interface IArticle {
  children: React.ReactNode;
}

export const Article: FunctionComponent<IArticle> = ({ children }) => (
  <ArticleContainer>{children}</ArticleContainer>
);
