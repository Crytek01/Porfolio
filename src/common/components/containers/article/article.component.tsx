import styled from 'styled-components';

export const ArticleContainer = styled.article`
  padding: ${({ theme: { spacing } }) => spacing[6]} 0
    ${({ theme: { spacing } }) => spacing[10]} 0;
`;
