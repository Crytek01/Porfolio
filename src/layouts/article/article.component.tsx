import { FunctionComponent } from 'react';

import { ArticleContainer } from './article.styles';

interface IArticle {
  children: React.ReactNode;
  className?: string;
}

export const Article: FunctionComponent<IArticle> = ({
  children,
  className,
}) => <ArticleContainer className={className}>{children}</ArticleContainer>;
