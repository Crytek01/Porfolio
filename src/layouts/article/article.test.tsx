import { shallowRender } from '@common/utils/test/shallowRender';
import { describe, expect, it } from 'vitest';

import { Article } from './article.component';

describe('Article component Test', () => {
  it('renders correctly', () => {
    const renderer = shallowRender(
      <Article>
        <div></div>
      </Article>
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
