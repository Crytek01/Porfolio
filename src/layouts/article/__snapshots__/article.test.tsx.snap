// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Article component Test > renders correctly 1`] = `
<styled.article>
  <div />
</styled.article>
`;
